core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpConnectionContextTest.java

```diff
@@ -289,6 +291,184 @@ public void testBadLineSyntax4() throws Exception {
+            String expected = "timestamp\tg\ta\twindspeed\n" +
+                    "1970-01-01T00:02:00.000000Z\t1\t0.1\t1.0\n" +
+                    "1970-01-01T00:03:00.000000Z\t4\t0.71\t67.72\n";
+            assertTable(expected, table);
+        });
+    }
+
+    @Test
+    public void testDuplicatedFieldsMapCountersInternational() throws Exception {
+        String table = "weather";
+
+        runInContext(() -> {
+            final TableStructureMock tableStructureMock = new TableStructureMock(engine.getConfiguration(), table);
+
+            try (final Path path = new Path()) {
+                try (final MemoryMARW ddlMem = Vm.getMARWInstance()) {
+                    final int status = engine.getStatus(lineTcpConfiguration.getCairoSecurityContext(), path, table);
+                    Assert.assertNotEquals(TableUtils.TABLE_EXISTS, status);
+
+                    engine.createTable(lineTcpConfiguration.getCairoSecurityContext(), ddlMem, path, tableStructureMock);
+                }
+            }
+
+            try (TableReader reader = engine.getReader(AllowAllCairoSecurityContext.INSTANCE, table)) {
+                final TableReaderMetadata metadata = reader.getMetadata();
+                Assert.assertEquals(1, metadata.getColumnCount());
+            }
+
+            recvBuffer =
+                    table + " г=2i,ветер=2.0,ветер=3.0 0\n" +
+                            table + " г=3i,а=0.4,а=0.5 60000000000\n" +
+                            table + " г=1i,а=0.1,ветер=1.0 120000000000\n" +
+                            table + " г=4i,а=0.71,ветер=67.72 180000000000\n";
+
+            handleContextIO();
+            Assert.assertFalse(disconnected);
+            closeContext();
+            String expected = "timestamp\tг\tа\tветер\n" +
+                    "1970-01-01T00:02:00.000000Z\t1\t0.1\t1.0\n" +
+                    "1970-01-01T00:03:00.000000Z\t4\t0.71\t67.72\n";
+            assertTable(expected, table);
+        });
+    }
+
+    @Test
+    public void testDuplicatedTagsTwoTables() throws Exception {
+        String table = "weather";
+        runInContext(() -> {
+            recvBuffer =
+                    "table val1=1,val2=2\n" +
+                            table + ",tag1=val1,tag2=val2 g=1i,a=0.1,windspeed=1.0 0\n" +
+                            table + ",tag1=val1,tag2=val2,tag1=val2 g=2i,windspeed=2.0 60000000000\n" +
+                            table + ",tag1=val1,tag2=val2,tag2=val1 g=3i,a=0.4 120000000000\n" +
+                            table + " g=4i,a=0.71,windspeed=67.72 180000000000\n";
+
+            handleContextIO();
+            Assert.assertFalse(disconnected);
+            closeContext();
+            String expected = "tag1\ttag2\tg\ta\twindspeed\ttimestamp\n" +
+                    "val1\tval2\t1\t0.1\t1.0\t1970-01-01T00:00:00.000000Z\n" +
+                    "\t\t4\t0.71\t67.72\t1970-01-01T00:03:00.000000Z\n";
+            assertTable(expected, table);
+        });
+    }
+
```

Comment:
Is there a test that covers the case when timestamp field value is provided two times: in the valueset and as the implicit value at the end of the line?